src/main/java/org/opensearch/neuralsearch/processor/NeuralQueryProcessor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.Map;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.neuralsearch.query.visitor.NeuralSearchQueryVisitor;
+import org.opensearch.search.pipeline.AbstractProcessor;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchRequestProcessor;
+
+/**
+ * Neural Search Query Request Processor
+ */
+public class NeuralQueryProcessor extends AbstractProcessor implements SearchRequestProcessor {
+
+    /**
+     * Key to reference this processor type from a search pipeline.
+     */
+    public static final String TYPE = "neural_query";
+
+    final String modelId;
+
+    final Map<String, Object> neuralFieldDefaultIdMap;
```

Comment:
private? Also, could we add a comment for this member?

Also, why is this map <String, Object> and not <String, String>

Reply:
Because [ConfigurationUtils](https://github.com/opensearch-project/OpenSearch/blob/d656e3db592f29466d35452867caa241f5429485/server/src/main/java/org/opensearch/ingest/ConfigurationUtils.java#L359) The class from Opensearch which is used to retrieve map from processor is generic and it Map<String, Object> defined in it. ConfigurationUtils is the designed to get values from the processor in OS. So I am using the same class